nncf/quantization/algorithms/accuracy_control/algorithm.py

```diff
@@ -0,0 +1,342 @@
+                                   validation_fn: Callable[[Any, Iterable[Any]], float],
+                                   max_drop: float = 0.01,
+                                   preset: QuantizationPreset = QuantizationPreset.PERFORMANCE,
+                                   target_device: TargetDevice = TargetDevice.ANY,
+                                   subset_size: int = 300,
+                                   fast_bias_correction: bool = True,
+                                   model_type: Optional[ModelType] = None,
+                                   ignored_scope: Optional[IgnoredScope] = None) -> TModel:
+    """
+    Common implementation of the `nncf.quantize_with_accuracy_control()` method.
+    """
+    # Step 0: Quantize provided model.
+    quantized_model = quantize(model, calibration_dataset, preset, target_device, subset_size,
+                               fast_bias_correction, model_type, ignored_scope)
+
+    # Backends
+    backend = get_backend(model)
+    algo_backend = get_algo_backend(backend)
+
+    initial_metric = validation_fn(algo_backend.prepare_for_inference(model),
+                                   validation_dataset.get_data())
+    nncf_logger.info(f'Metric of initial model: {initial_metric}')
+
+    quantized_metric = validation_fn(algo_backend.prepare_for_inference(quantized_model),
+                                     validation_dataset.get_data())
+    nncf_logger.info(f'Metric of quantized model: {quantized_metric}')
+
+    accuracy_aware_loop = AccuracyAwareLoop(is_native=False)
+    return accuracy_aware_loop.restore_accuracy(model, initial_metric,
+                                                quantized_model, quantized_metric,
+                                                validation_dataset, validation_fn, max_drop)
+
+
+def _create_message(nodes: Iterable[NNCFNode]) -> str:
+    names = [f'\t{x.node_name}' for x in nodes]
+    return '\n'.join(names)
+
+
+class AccuracyAwareLoop:
+
+    def __init__(self,
+                 ranking_subset_size: int = 300,
+                 max_num_iterations: int = sys.maxsize,
+                 is_native: bool = True):
+        """
+        :param ranking_subset_size:
+        :param max_num_iterations:
+        :param is_native:
+        """
+        self.ranking_subset_size = ranking_subset_size
+        self.max_num_iterations = max_num_iterations
+        # TODO(andrey-churkin): Should be removed when native implementation
+        # will become the main one.
+        self.is_native = is_native
+
+    def restore_accuracy(self,
+                         initial_model: TModel,
+                         initial_metric: float,
+                         quantized_model: TModel,
+                         quantized_metric: float,
+                         validation_dataset: Dataset,
+                         validation_fn: Callable[[Any, Iterable[Any]], float],
+                         max_drop: float = 0.01) -> TModel:
```

Comment:
Why not set `max_drop` as the `self.` variable? It can't be changed during the pipeline.

Reply:
Done